proposals/interface-types/working-notes/Instructions.md

```diff
@@ -0,0 +1,252 @@
+# Interface instructions
+
+This proposal defines new instructions that operate over interface value types. These instructions have the same form as core instructions; they push and pop values on an execution stack and support single pass validation.
+
+The new interface instructions are summarized below.
+
+```
+#interface-instr ::=
+  | bool.lift_iMM
+  | bool.lower_iMM
+
+  | sNN.lift_iMM
+  | sNN.lower_iMM
+
```

Comment:
I have been using the convention of 
tt-to-tt 
for both lifting and lowering instructions. This reflects the core functionality - based on coercion; rather than the intended purpose.
In general, you need to mention two types; and lifting/lowering is not required to be explicit.

Reply:
i32-to-u64 says what the instruction does, not what it is for. Many of the instructions are coercion instructions: coercing values from one type to another.
For naming instructions, IMO it is better to focus on what the instruction does.
Apart from anything else, if you use lift/lower, you will see a large number of lifts and lowers in your code.
But, the main reason is that these instructions may be used in unexpected ways; i.e., not for the purpose you think they should be used for.

Reply:
We decided in the meeting that we'll go with `$dst.$verb_$src` form for now. I wasn't following that strictly, but have now updated to it. Tried to also pick reasonable verbs.